### ISSUE TITLE
MySQL backend support

### ISSUE THREAD

**Issue by valderman** [valderman/selda#22]
Currently, a few things are missing to be able to support a MySQL backend:
* Support for `?` as a parameter placeholder.
* Probably more.

This is probably a good time to restructure the pretty-printer to be explicitly parameterized over some configuration, instead of over different functions in different places.

**Comment by winterland1989** [comment]
For now my only request is to custom placeholder ; )

**Comment by valderman** [comment]
@winterland1989 93bd5d0b22968af48beba8f60bfeb3c072cbf2f6 moves the pretty-printer configuration into a single type, and adds support for custom placeholders.

**Comment by winterland1989** [comment]
Nice! Would you mind keeping this issue open? I'll update info if i come across further problem.

**Comment by valderman** [comment]
Indeed, I won't feel comfortable closing this until we actually have MySQL up and working.

**Comment by winterland1989** [comment]
@kuribas  Can you speak more specifically about what the bug is?

**Comment by kuribas** [comment]
Sorry, it was a bug in my own code, when copy pasting your functions.
